src/pudl/transform/ferc1.py

```diff
@@ -1047,7 +1057,13 @@ def unstack_balances_to_report_year_instant_xbrl(
         if params is None:
             params = self.params.unstack_balances_to_report_year_instant_xbrl
         if params.unstack_balances_to_report_year:
-            df = unstack_balances_to_report_year_instant_xbrl(df, params=params)
+            df = unstack_balances_to_report_year_instant_xbrl(
+                df,
+                params=params,
+                primary_key_cols=self.source_table_primary_key(
+                    source_ferc1=SourceFerc1.XBRL
+                ),
+            )
```

Comment:
I think this is generally fine! and certainly good enough to merge in. In fact I think  we should!! 

but also I'm curious if instead of adding this as an argument if we can add it to `params` within this method. iirc you (reasonably!) made the params base class immutable. Which i think is the correct default. There are several instances of parameter elements that we could generate within the class itself which would enable us to stop hard coding things like the source table.